### PR TITLE
fix issue #530 SBR can not be deleted after the bound application has been deleted

### PR DESCRIPTION
**Pull request by qibobo**

### CODE REVIEW

**Comment by Avni-Sharma** [comment]
Hi @qibobo Thanks for the PR. Can we have an additional test in the reconciler_test.go also to see the reconcile activity on a deleted application?

File: pkg/controller/servicebindingrequest/binder_test.go [hunk 1]
```diff
@@ -480,3 +483,61 @@ func Test_extraFieldsModifier(t *testing.T) {
 	})
 
 }
+
+func TestBindAndUnbind(t *testing.T) {
+
+	t.Run("bind and unbind", func(t *testing.T) {
```
Comment (pedjak) [inline pkg/controller/servicebindingrequest/binder_test.go:489]: can you please share more infos what we are testing here, given the scope of PR? And update the test name accordingly: what is the context and the expected outcome?

File: pkg/controller/servicebindingrequest/binder_test.go [hunk 2]
```diff
@@ -480,3 +483,61 @@ func Test_extraFieldsModifier(t *testing.T) {
...
+		binder := newBinder(
+			context.TODO(),
+			f.FakeDynClient(),
+			sbr,
+			[]string{},
+			testutils.BuildTestRESTMapper(),
+		)
+		require.NotNil(t, binder)
+
+		_, err := binder.search()
```
Comment (pedjak) [inline pkg/controller/servicebindingrequest/binder_test.go:505]: `binder.search()` is invoked in `binder.bind()`, so not sure if we really needed here?

File: pkg/controller/servicebindingrequest/reconciler.go [hunk 3]
```diff
@@ -108,7 +108,7 @@ func (r *reconciler) Reconcile(request reconcile.Request) (reconcile.Result, err
 	// fetch and validate namespaced ServiceBindingRequest instance
 	sbr, err := r.getServiceBindingRequest(request.NamespacedName)
 	if err != nil {
-		if errors.Is(err, applicationNotFound) {
+		if k8sapierrors.IsNotFound(err) {
```
Comment (pedjak) [inline pkg/controller/servicebindingrequest/reconciler.go:111]: We need an e2e tests checking this.

File: pkg/controller/servicebindingrequest/binder_test.go (hunk 1 above)
Comment (qibobo) [reply pkg/controller/servicebindingrequest/binder_test.go]: Previously there is no UT for bind and unbind. I add them and have added more descriptions.

File: pkg/controller/servicebindingrequest/binder_test.go (hunk 2 above)
Comment (qibobo) [reply pkg/controller/servicebindingrequest/binder_test.go:505]: Add a comment for this. We need to ensure there is no error in search().

**Comment by qibobo** [comment]
@Avni-Sharma @pedjak add the UT and description.

**Comment by qibobo** [comment]
issue has been fixed.

**Pull request closed without merging**